src/test/java/com/emc/object/s3/S3JerseyClientTest.java

```diff
@@ -199,6 +210,228 @@ public void testCreateEncryptedBucket() {
+        ObjectLockConfiguration objectLockConfig = client.getObjectLockConfiguration(bucketName);
+        Assert.assertEquals("Enabled", objectLockConfig.getObjectLockEnabled());
+        client.deleteBucket(bucketName);
+    }
+
+    @Test
+    public void testSetObjectLockConfiguration() {
+        Assume.assumeTrue("Skip Object Lock related tests for non IAM user.", testIAM);
+
+        String bucketName = getTestBucket();
+        ObjectLockConfiguration objectLockConfig = new ObjectLockConfiguration().withObjectLockEnabled(ObjectLockConfiguration.ObjectLockEnabled.Enabled);
+        DefaultRetention defaultRetention = new DefaultRetention().withMode(ObjectLockRetentionMode.GOVERNANCE).withDays(2);
+        objectLockConfig.setRule(new ObjectLockRule().withDefaultRetention(defaultRetention));
+        try {
+            client.setObjectLockConfiguration(bucketName, objectLockConfig);
+        } catch (S3Exception e) {
+            //Object Lock configuration cannot be set on existing bucket without ObjectLock being enabled.
+            Assert.assertEquals(409, e.getHttpCode());
+            Assert.assertEquals("InvalidBucketState", e.getErrorCode());
+        }
+
+        client.enableObjectLock(bucketName);
+        client.setObjectLockConfiguration(bucketName, objectLockConfig);
+        ObjectLockConfiguration objectLockConfig_verify = client.getObjectLockConfiguration(bucketName);
+
+        Assert.assertEquals(objectLockConfig.getObjectLockEnabled(), objectLockConfig_verify.getObjectLockEnabled());
+        Assert.assertEquals(defaultRetention.getMode(), objectLockConfig_verify.getRule().getDefaultRetention().getMode());
+        Assert.assertEquals(defaultRetention.getDays(), objectLockConfig_verify.getRule().getDefaultRetention().getDays());
+        Assert.assertEquals(defaultRetention.getYears(), objectLockConfig_verify.getRule().getDefaultRetention().getYears());
+    }
+
+    @Test
+    public void testDeleteObjectWithLegalHoldNotAllowed() throws Exception {
+        Assume.assumeTrue("Skip Object Lock related tests for non IAM user.", testIAM);
+
+        String bucketName = getTestBucket();
+        String key = "testObject_DeleteWithLegalHold";
+        client.enableObjectLock(bucketName);
+        ObjectLockLegalHold objectLockLegalHold = new ObjectLockLegalHold().withStatus(ObjectLockLegalHold.Status.ON);
+        PutObjectRequest putObjectRequest = new PutObjectRequest(bucketName, key,"test Delete With LegalHold Not Allowed")
+                .withObjectLockLegalHold(objectLockLegalHold);
+        client.putObject(putObjectRequest);
+        String versionId = client.listVersions(bucketName, key).getVersions().get(0).getVersionId();
+
+        Assert.assertEquals("ON", client.getObjectLegalHold(new GetObjectLegalHoldRequest(bucketName,key).withVersionId(versionId)).getStatus());
+        try {
+            client.deleteVersion(bucketName, key, versionId);
```

Comment:
add `Assert.fail` here - if no exception is thrown, that should fail the test

Reply:
Fixed.